### PR TITLE
Clear code coverage upon file save

### PR DESCRIPTION
**Pull request by alanlal**
PR for  [#2551](https://github.com/microsoft/vscode-go/issues/2551https://github.com/microsoft/vscode-go/issues/2551)

Update `goMain.ts` 

Added a new function to `goCover.ts` , `removeCodeCoverageOnFileSave` to handle File Save for clearing code coverage.

### CODE REVIEW

**Comment by alanlal** [comment]
Hi @ramya-rao-a,

I'm having trouble finding a solution to this. Could you help me figure out the areas that I should check out? I looked at the [ vscode api docs ](https://code.visualstudio.com/api/references/vscode-api#working-with-extensions-articles); but I couldn't find a file save event listener that takes arguments that contains file change information.

File: src/goMain.ts [hunk 1]
```diff
@@ -336,7 +336,7 @@ export function activate(ctx: vscode.ExtensionContext): void {
 	ctx.subscriptions.push(lintDiagnosticCollection);
 	vetDiagnosticCollection = vscode.languages.createDiagnosticCollection('go-vet');
 	ctx.subscriptions.push(vetDiagnosticCollection);
-	vscode.workspace.onDidChangeTextDocument(removeCodeCoverageOnFileChange, null, ctx.subscriptions);
+	vscode.workspace.onDidSaveTextDocument(removeCodeCoverageOnFileSave, null, ctx.subscriptions);
```
Comment (alwindoss) [inline src/goMain.ts]: Shouldn't this line go into the function `addOnSaveTextDocumentListeners`

**Comment by ramya-rao-a** [comment]
@alanlal Looks like there is no good way to get the changes done to a document when it is saved.

**Comment by ramya-rao-a** [comment]
Closing this PR as there hasnt been any updates to it since last June

**Comment by alwindoss** [comment]
@ramya-rao-a I thought someone had raised a PR for the same. If it's not so let me give this a shot.

However, why are we closing this issue, it is not fixed right?

**Comment by ramya-rao-a** [comment]
@alwindoss, The issue is not being closed, only the PR is being closed. The issue #2551 is still open and I will follow up on the alternate PR #2853 that builds on your work and addresses the comments I had made.

**Pull request closed without merging**